tools/kicad_rs/src/parser.rs

```diff
@@ -128,19 +131,24 @@ fn parse_schematic(p: &Path, id: String) -> Result<Schematic, Box<dyn Error>> {
                 &m,
             ).or_empty_str().is_true_like()
         {
-            // Validate that reference and package aren't empty
-            if c.reference.is_empty() {
-                return Err(Box::new(errorf(&format!(
-                    "{}: Component.reference is a mandatory field",
-                    &comp.name
-                ))));
-            }
-            if c.package.is_empty() {
-                return Err(Box::new(errorf(&format!(
-                    "{}: Component.package is a mandatory field",
-                    &comp.name
-                ))));
+            // Validate that required fields aren't empty
+            let mut req_fields = HashMap::new();
+            req_fields.insert("reference", &c.reference);
+            req_fields.insert("footprint_library", &c.footprint_library);
+            req_fields.insert("footprint_name", &c.footprint_name);
+            req_fields.insert("symbol_library", &c.symbol_library);
+            req_fields.insert("symbol_name", &c.symbol_name);
```

Comment:
Does this block do the validation?

Reply:
Looks like the fields are inserted into the `HashMap` regardless of whether they are empty or not.

Reply:
Yes, but then they are validated not to be empty in the for loop directly after.

Reply:
Maybe remove the comment that's the source of the confusion. It seems to say the same thing as the comment on line 142, which is more appropriately placed.
```suggestion
            let mut req_fields = HashMap::new();
            req_fields.insert("reference", &c.reference);
            req_fields.insert("footprint_library", &c.footprint_library);
            req_fields.insert("footprint_name", &c.footprint_name);
            req_fields.insert("symbol_library", &c.symbol_library);
            req_fields.insert("symbol_name", &c.symbol_name);
```

Reply:
I've updated the comment to better reflect what's happening here.